app/views/recipes/_form.html.erb

```diff
@@ -133,4 +133,39 @@
     </div>
   </div>
 
-<% end %>
\ No newline at end of file
+  <div class='form-group input-switch' ng-switch='recipe_<%= @recipe.id %>_file' ng-init='recipe_<%= @recipe.id %>_file=<%= @recipe.file.present? %>'>
+    <div ng-hide='recipe_<%= @recipe.id %>_file' ng-click="recipe_<%= @recipe.id %>_file=true" class='media col-xs-4 col-xs-offset-4'>
+      <div class='media-left'>
+        <span class="fa-stack">
+          <i class="fa fa-stack-2x fa-circle-thin"></i>
+          <i class="fa fa-stack-1x fa-cloud-upload"></i>
+        </span>
+      </div>
+      <div class='media-body'>
+        <span class='help-text'>Upload a file</span>
+        <ul class='icon-list list-inline'>
+          <li><i class='fa fa-file-xlsx-o fa-3x'></i></li>
```

Comment:
Can we use the PDF icon here as well (though that may be too much – so maybe just the PDF and then then generic file icon?). Here's the list of all the file icons as they are given now: http://fortawesome.github.io/Font-Awesome/icons/#file-type

Leave it up to you to try them out and see what you think looks best!